### ISSUE TITLE
Document how to use as a JS library with `require("rainbow-js-arc")`

### ISSUE THREAD

**Issue by rocketnia** [arclanguage/rainbow-js#8]
We should document how people can write their own I/O systems for Rainbow.js the way `index-first.js`, `index-last.js`, and `rainbow-node-src.js` do.

**Comment by hw-a** [comment]
any progress or demo/examples on this? thanks

**Comment by rocketnia** [comment]
No, there hasn't been any work on this since creating the issue. The best "documentation" right now is the code in the repo (`index-first.js`, `index-last.js`, `rainbow-node.js`).

**Comment by rocketnia** [comment]
To make it slightly easier to channel effort on this whenever someone is motivated to work on it, I'll note this issue is related to:

- https://github.com/arclanguage/anarki/issues/227

**Comment by hw-a** [comment]
i see. thanks
